frontend/src/main/web/lib/stores/GlossaryStore.js

```diff
@@ -0,0 +1,522 @@
+      .set('Accept', 'application/json')
+      .end(function (err, res) {
+        if(err != null && err.error === true) {
+          console.error(url, err);
+        }
+        if(res != null) {
+          if (res.error) {
+            console.error(url, res.status, res.error.toString());
+            reject(Error(res.error.toString()));
+          } else {
+            resolve(res['body']);
+          }
+        }
+        resolve(null);
+      })
+  });
+}
+function saveGlossary(data) {
+  var entry = {
+    resId: '', pos: data.pos, description: data.description,
+    srcTerm: GlossaryHelper.generateSrcTerm(data.srcLocaleId)
+  };
+  entry.srcTerm.content = data.term;
+  var glossary = GlossaryHelper.generateGlossaryDTO(entry);
+  return saveOrUpdateGlossary(glossary);
+}
+
+function updateGlossary(entry) {
+  entry.status.isSaving = true;
+  var glossary = GlossaryHelper.generateGlossaryDTO(entry);
+  return saveOrUpdateGlossary(glossary);
+}
+
+function saveOrUpdateGlossary(glossary) {
+  var url = Configs.baseUrl + "/glossary/" + Configs.urlPostfix;
+  return new Promise(function(resolve, reject) {
+    Request.post(url)
+      .set('Content-Type', 'application/json')
+      .set('Accept', 'application/json')
+      .send(glossary)
+      .end(function (err, res) {
+        if(err != null && err.error === true) {
+          console.error(url, err);
+        }
+        if(res != null) {
+          if (res.error) {
+            console.error(url, res.status, res.error.toString());
+            reject(Error(res.error.toString()));
+          } else {
+            resolve(res['body']);
+          }
+        }
+        resolve(null);
+      })
+  });
+}
+
+function uploadFile(data) {
+  var url = Configs.baseUrl + "/glossary/upload",
+    uploadFile = data.uploadFile;
+
+  _state['uploadFile']['status'] = 0;
+
+  return new Promise(function(resolve, reject) {
+    Request.post(url)
+      .attach('file', uploadFile.file, uploadFile.file.name)
+      .field('fileName', uploadFile.file.name)
+      .field('srcLocale', data.srcLocale)
+      .field('transLocale', data.uploadFile.transLocale)
+      .set('Accept', 'application/json')
+      .end(function (err, res) {
+        _state['uploadFile']['status'] = -1;
+        _state['uploadFile'].show = false;
+        _state['uploadFile'].transLocale = null;
+        _state['uploadFile'].file = null;
+
+        if(err != null && err.error === true) {
+          console.error(url, err);
+        }
+        if(res != null) {
+          if (res.error) {
+            console.error(url, res.status, res.error.toString());
+            reject(Error(res.error.toString()));
+          } else {
+            resolve(res['body']);
+          }
+        }
+        resolve(null);
+      })
+      .on('progress', function(e) {
+        _state['uploadFile']['status'] = e.percent;
+      });
+  });
+}
+
+function processDelete(serverResponse) {
+  console.debug('Glossary entry deleted');
+}
```

Comment:
Please add a comment to say that it needs notification feature before it can be implemented.